pkg/syncer/metadatasyncer.go

```diff
@@ -2382,6 +2382,56 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 // Vanills k8s and supervisor cluster.
 func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *metadataSyncInformer) {
 	log := logger.GetLogger(ctx)
+	if isPodVMOnStretchSupervisorFSSEnabled {
+		volumeInfo, err := volumeInfoService.GetVolumeInfoForVolumeID(ctx, pv.Spec.CSI.VolumeHandle)
+		if err != nil {
+			log.Errorf("failed to fetch CnsVolumeInfo CR. Error: %+v", err)
+			return
+		}
+		restConfig, err := config.GetConfig()
+		if err != nil {
+			log.Errorf("failed to fetch kubernetes config. Error: %+v", err)
+			return
+		}
+		cnsOperatorClient, err := k8s.NewClientForGroup(ctx, restConfig, cnsoperatorv1alpha1.GroupName)
+		if err != nil {
+			log.Errorf("failed to create CNSOperator client. Error: %+v", err)
+			return
+		}
+
+		// Fetch StoragePolicyUsage instance for storageClass associated with the volume.
+		storagePolicyUsageInstanceName := volumeInfo.Spec.StorageClassName + "-" +
+			storagepolicyusagev1alpha1.NameSuffixForPVC
+		storagePolicyUsageCR := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+		err = cnsOperatorClient.Get(ctx, k8stypes.NamespacedName{
+			Namespace: volumeInfo.Spec.Namespace,
+			Name:      storagePolicyUsageInstanceName},
+			storagePolicyUsageCR)
+		if err != nil {
+			log.Errorf("failed to fetch %s instance with name %q from supervisor namespace %q. Error: %+v",
+				storagepolicyusagev1alpha1.CRDSingular, storagePolicyUsageInstanceName,
+				volumeInfo.Spec.Namespace, err)
+			return
+		}
+
+		// Decrease the used capacity in StoragePolicyUsage instance as we are deleting the volume.
+		patchedStoragePolicyUsageCR := storagePolicyUsageCR.DeepCopy()
+		patchedStoragePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used.Sub(*volumeInfo.Spec.Capacity)
+		err = PatchStoragePolicyUsage(ctx, cnsOperatorClient, storagePolicyUsageCR,
+			patchedStoragePolicyUsageCR)
+		if err != nil {
+			log.Errorf("updateStoragePolicyUsage failed. err: %v", err)
+			return
+		}
+		log.Infof("Successfully decreased the used capacity to zero(0) for StoragePolicyUsage: %q in namespace: %q",
```

Comment:
Every time there is delete, you will just decrease the capacity of the PVC right? Won't be zero always

Reply:
Right :) . Good catch. Updated it.